File: lib/bodies/spare.dart
class Spare {
  final String type;
  final String number;
  final int weight;

  Spare(this.type, this.number, this.weight);

  Map<String, dynamic> toJson() {
    return {
      "Type": type,
      "Number": number,
      "Weight": weight,
    };
  }
}

File: core/spare.go
package core

import "github.com/louisevanderlith/husk/validation"

type Spare struct {
	Number    string
	Type      string
	Weight    int
	AppliesTo []string `hsk:"null"`
}

func (o Spare) Valid() error {
	return validation.Struct(o)
}
